Automattic/pocket-casts-android
# Add episode basic authentication support

## Description
Downloading or streaming episodes fail if there is a basic authentication username and password in the url. For example https://username:password@website.com/episode.m4a. To fix this I have added the HTTP header 'Authorization' if the URL contains a username and password. 

Slack discussion p1731364537074479/1731336095.261839-slack-C02A333D8LQ

## Testing Instructions
1. Add the podcast from the Slack discussions above
2. Download an episode
3. ✅ Verify it successfully downloads
4. Stream an episode
5. ✅ Verify it successfully plays
6. Go to Profile -> Settings -> Beta features and turn off 'Support episode basic authentication'
7. ✅ Verify you can't download or stream any episodes

## Checklist
 
#### I have tested any UI changes...
